clients/python-wrapper/lakefs/exceptions.py

```diff
@@ -63,7 +80,32 @@ class PermissionException(NotAuthorizedException, PermissionError):
     """
 
 
-class RepositoryNotFoundException(NotFoundException):
+class RefNotFoundException(NotFoundException):
+    """
+    Raised when the reference could not be found in the lakeFS server
+    """
+
+
+_STATUS_CODE_TO_EXCEPTION = {
+    http.HTTPStatus.UNAUTHORIZED.value: NotAuthorizedException,
+    http.HTTPStatus.FORBIDDEN.value: ForbiddenException,
+    http.HTTPStatus.NOT_FOUND.value: NotFoundException,
+    http.HTTPStatus.METHOD_NOT_ALLOWED.value: UnsupportedOperationException,
+    http.HTTPStatus.CONFLICT.value: ConflictException,
+}
+
+
+@contextmanager
+def api_exception_handler(custom_handler: Optional[Callable[[LakeFSException], None]] = None):
     """
-    Raised when the currently used repository object no longer exists in the lakeFS server
+    Contexts which converts lakefs_sdk API exceptions to LakeFS exceptions and handles them
+    :param custom_handler: Optional handler which can be used to provide custom behavior for specific exceptions
     """
+    try:
+        yield
+    except lakefs_sdk.ApiException as e:
+        lakefs_ex = _STATUS_CODE_TO_EXCEPTION.get(e.status, ServerException)(e.status, e.reason)
+        if custom_handler is None:
+            raise lakefs_ex from e
+
+        custom_handler(lakefs_ex)
```

Comment:
I don't understand exactly what you mean. Anyway your suggestion requires us to return a return value from the custom handler. I think allowing the handler to have the ability to just perform any required logic is better

Reply:
Sure, my suggestion is to enable that but also control if we like the default behaviour.
For example the case for create or get repository - we can capture specific exception by the custom handler, and return false and let the exception handler raise all the standard exceptions for the other cases.

The current implementation passes the exception `e` to the custom handler which rise it but will not be able to `raise lakefs_ex from e`.

Reply:
Thanks for the explanation - I've made some modifications. Instead of returning a bool the custom handler will return an exception (or None). I think this solution is more versatile